### PR TITLE
Add documentation examples for LineEdit class

### PR DESCRIPTION
**Pull request by brandonlamb**
Add more documentation code examples for the LineEdit classes to show
more full examples.

### CODE REVIEW

File: doc/base/classes.xml [hunk 1]
```diff
@@ -20033,6 +20033,8 @@
 			</return>
 			<description>
 				Return the maximum amount of characters the [LineEdit] can edit. If 0 is returned, no limit exists.
+        
+        Once this limit is reached, no more text will be added to the LineEdit.
```
Comment (akien-mga) [inline doc/base/classes.xml]: Indentation issue, it should use tabs, not spaces.

File: doc/base/classes.xml [hunk 2]
```diff
@@ -20058,13 +20060,41 @@
 			</return>
 			<description>
 				Return the text in the [LineEdit].
+        
+        [code]
```
Comment (akien-mga) [inline doc/base/classes.xml]: You can't use `[code]` for that, it's meant for inline code. You can however use `[codeblock]`.

File: doc/base/classes.xml [hunk 3]
```diff
@@ -20058,13 +20060,41 @@
...
+        extends Node2D
+        
+        onready var __history = find_node("RichTextLabel")
```
Comment (akien-mga) [inline doc/base/classes.xml]: Prefixing variables with two underscores is not idiomatic GDScript.

File: doc/base/classes.xml [hunk 4]
```diff
@@ -20058,13 +20060,41 @@
...
+        onready var __input = find_node("LineEdit")
+        
+        func _ready():
```
Comment (akien-mga) [inline doc/base/classes.xml]: No need to add a callback that is not relevant in the example.

File: doc/base/classes.xml [hunk 5]
```diff
@@ -20124,12 +20184,32 @@
...
 			<description>
 				Set the maximum amount of characters the [LineEdit] can edit, and cropping existing text in case it exceeds that limit. Setting 0 removes the limit.
+        
+        [code]
+        extends Node2D
+        
+        onready var __input = find_node("LineEdit")
+        
+        func _ready():
+          __input.set_max_length(140) # set input to max 140 characters for a tweet
```
Comment (akien-mga) [inline doc/base/classes.xml]: I don't think such examples are useful at all, it's pretty obvious from the function definition that it's how they're meant to be used. If we start adding code snippets with
```
extends Node2D

onready var somestuff

func _ready():
    use_function_described_above(some_arg)  # do what the function says it does
```
we're not really teaching people anything meaningful, and we're bloating the docs with lots of noise.

**Review by akien-mga — CHANGES_REQUESTED** [review]
Thanks for your contribution, but I don't see those additions as useful to the Godot documentation.

The examples are trivial and giving full scripts to showcase the utilisation of simple functions like `LineEdit.get_text()` is really overkill IMO.

I think examples should only be given for functions that need some context to be used. The purpose of documentation is not that you can learn how to use Godot just by reading the documentation of a single function, it's assumed that users know how to call a simple function to do what its documentation says. If they don't, they should read the tutorials and not the classref.

**Comment by brandonlamb** [comment]
Good morning!

- I am interested in any other feedback from others
- I added these because I myself was looking for more contextual usage of these and came back around afterwards thinking I could contribute
- Is it possible for documentation to be "bloated"? I would think the more the better, I mean it is documentation afterall
- One good thing about PHP documentation as a comparison, is all the example usages posted
- Maybe there could be a single examples section for a class rather than each method?
- Remember, documentation is not for godot developers but users
- Your "trivial" may be someone else's "aha moment"

If these are not desired, no offense taken. Maybe a note should be added for potential contributors to know the desired documentation level, dos and donts?

File: doc/base/classes.xml (hunk 1 above)
Comment (brandonlamb) [inline doc/base/classes.xml]: Ahhh, good to know

**Comment by vnen** [comment]
> Is it possible for documentation to be "bloated"? [...] One good thing about PHP documentation as a comparison, is all the example usages posted

Yes, it is. A whole wall of text for a simple function would make it seem more complex than it is and also will start being not read at all (so it's like the same as having no docs). The major problem here is that this all appear on a single page in the website and in the editor. Having long function descriptions would make navigation horrible.

PHP has one page per function, which certainly increases the space you can have for each one, adding usage examples and such. It has been proposed (godotengine/godot-docs#133) but until it's done the descriptions should remain short.

When we have one page per function, then usage examples will have their place. But I agree that adding `function_called(arg) # makes what the description says` does not add anything meaningful. For instance, your examples start with `extends Node2D`, but the example itself could happen in any of kind of node. If these are intended to be "copy & paste & run" examples, they depend too much on the implicit scene tree to "just work".

> Maybe a note should be added for potential contributors to know the desired documentation level, dos and donts.

This might be improved. We do want better documentation (though "better" might be subjective). At least we want to fill the undocumented classes and functions. However, you need to note that function usage examples in the existing docs are very rare (if there are any) and you just assumed they would be good without asking first. Had you opened an issue beforehand, this lost work could be avoided.

**Pull request closed without merging**